tests/Integration/Behaviour/Features/Context/Domain/OrderFeatureContext.php

```diff
@@ -1561,10 +1595,11 @@ private function getOrderDetailFromOrder(
         ?string $combinationName = null,
         ?int $orderInvoiceId = null
     ): array {
-        $product = $this->getProductByName($productName);
+        $order = new Order(SharedStorage::getStorage()->get($orderReference));
+        $product = $this->getProductByNameAndOrder($productName, $order);
```

Comment:
That's a good idea to use this one with the search restricted for Order it allows to validate the Search query indirectly However I don't understand why you improved this method here, but decided not to rely on it in `editProductsToOrderWithFollowingDetails`